lib/ar_lazy_preload/active_record/relation.rb

```diff
@@ -82,6 +80,16 @@ def lazy_preload_values
 
     private
 
+    def perform_preloading(records, associations)
+      if ::ActiveRecord::VERSION::MAJOR >= 7
+        ActiveRecord::Associations::Preloader.new(
+          records: records, associations: associations
+        ).call
+      else
+        ActiveRecord::Associations::Preloader.new.preload(records, associations)
+      end
```

Comment:
**Question:** Maybe it's easier/better to have 2 different classes for version 7 and below? So we can include in appropriate version only what we need.